src/main/index.ts

```diff
@@ -128,9 +129,67 @@ log.info("Application launched");
 // Enforce sandbox on all renderers
 app.enableSandbox();
 
-// Disabling the application menu improves performance, but we're disabling it to patch out default hotkeys for now
-// TODO: Come back to this at a future point in time to determine the necessary need for an application menu
-Menu.setApplicationMenu(null);
+const application: MenuItemConstructorOptions = {
+  label: "Application",
+  submenu: [
+    {
+      label: "About Application",
+      role: "about"
+    },
+    {
+      type: "separator"
+    },
+    {
+      label: "Quit",
+      accelerator: "Command+Q",
+      click: () => {
+        app.quit();
+      }
+    }
+  ]
+};
+
+const edit: MenuItemConstructorOptions = {
+  label: "Edit",
+  submenu: [
+    {
+      label: "Undo",
+      accelerator: "CmdOrCtrl+Z",
+      role: "undo"
+    },
+    {
+      label: "Redo",
+      accelerator: "Shift+CmdOrCtrl+Z",
+      role: "redo"
+    },
+    {
+      type: "separator"
+    },
+    {
+      label: "Cut",
+      accelerator: "CmdOrCtrl+X",
+      role: "cut"
+    },
+    {
+      label: "Copy",
+      accelerator: "CmdOrCtrl+C",
+      role: "copy"
+    },
+    {
+      label: "Paste",
+      accelerator: "CmdOrCtrl+V",
+      role: "paste"
+    },
+    {
+      label: "Select All",
+      accelerator: "CmdOrCtrl+A",
+      role: "selectAll"
+    }
+  ]
+};
+
+const template = [application, edit];
+Menu.setApplicationMenu(Menu.buildFromTemplate(template));
```

Comment:
Since other operating systems don't need an application menu, and at least on windows, we don't even show one I would platform guard setting an application menu to mac and explicitly set it to null on other operating systems.